src/annotator/plugin/pdf.js

```diff
@@ -0,0 +1,123 @@
+import debounce from 'lodash.debounce';
+
+import * as pdfAnchoring from '../anchoring/pdf';
+import RenderingStates from '../pdfjs-rendering-states';
+// @ts-expect-error - Plugin module is CoffeeScript
+import Plugin from '../plugin';
+
+import PDFMetadata from './pdf-metadata';
+
+/**
+ * @typedef {import('../../types/annotator').Anchor} Anchor
+ * @typedef {import('../../types/annotator').Annotator} Annotator
+ * @typedef {import('../../types/annotator').HypothesisWindow} HypothesisWindow
+ */
+
+export default class PDF extends Plugin {
+  constructor(element, config) {
+    super(element, config);
+
+    /**
+     * The `Guest` instance for the current document.
+     * nb. This is initialized by `Guest` before it calls `pluginInit`.
+     *
+     * @type {Annotator}
+     */
+    this.annotator = /** @type {any} */ (null);
+
+    const window_ = /** @type {HypothesisWindow} */ (window);
+    this.pdfViewer = window_.PDFViewerApplication.pdfViewer;
```

Comment:
One intentional change in this port is that several of the properties that were initialized in `pluginInit` are now initialized in the constructor instead.

This is in preparation for removing the `pluginInit` method in future, since there is no real need any more to have this two-step initialization process.